lib/core/test/unit/Cardano/Wallet/Primitive/ModelSpec.hs

```diff
@@ -452,10 +456,11 @@ instance Arbitrary (WithPending WalletState) where
                 -- - Sometimes, we also add a withdrawal by creating another
                 --   change output and an explicit withdrawal in the
                 --   transaction.
+                let tokens = TB.fromCoin $ simulateFee $ txOutCoin out
                 let pending = withWithdrawal $ Tx
                         { txId = arbitraryHash
-                        , resolvedInputs = [(inp, coin out)]
-                        , outputs = [out { coin = simulateFee (coin out) }]
+                        , resolvedInputs = [(inp, txOutCoin out)]
+                        , outputs = [out {tokens}]
```

Comment:
This needs checking. It looks like the call to `simulateFee` was missed out here.

Reply:
Ah, turns out it wasn't missed out after all:

```
                let tokens = TB.fromCoin $ simulateFee $ txOutCoin out
                let pending = withWithdrawal $ Tx 
                        { txId = arbitraryHash
                        , resolvedInputs = [(inp, txOutCoin out)] 
                        , outputs = [out {tokens}]
                        , withdrawals = mempty
                        , metadata = Nothing  
                        }
```